crucible-llvm/src/Lang/Crucible/LLVM/MemModel/Generic.hs

```diff
@@ -1366,6 +1433,49 @@ possibleAllocs n = helper . memAllocs
               AllocMerge (asConstantPred -> Just False) as1 as2 -> helper as2
               AllocMerge _ as1 as2 -> helper as1 ++ helper as2
 
+asMemAllocationArrayStore ::
```

Comment:
This could probably use a comment

Reply:
Agree, documentation here is important.  I'd also like a brief discussion of the performance implications of this @andreistefanescu, as we talked about in person the other day.  In case we later decide to go forward with the caching ideas we talked about, it would be good to record our thoughts on the high-level ideas.